### PR TITLE
Refactor part 5

### PR DESCRIPTION
**Pull request by koetsier**
More refactoring

- Remove Document#summary to remove dependency of Document on FinderPresenter 
- Move metadata related methods from FinderPresenter to Document so that Document is now the only place that knows about a Document's metadata
- Rewrite specs for ResultSetPresenter to use real instances instead of doubles. Made the tests more descriptive.
- Remove redundant rescue
- Document is now no longer dependent on FinderPresenter

Trello: https://trello.com/c/7mcVTnp1/978-refactor-finder-frontend-v

## Search page examples to sanity check:

- http://finder-frontend-pr-1333.herokuapp.com/search/all
- http://finder-frontend-pr-1333.herokuapp.com/search/research-and-statistics
- http://finder-frontend-pr-1333.herokuapp.com/search/news-and-communications?parent=%2Feducation&topic=c58fdadd-7743-46d6-9629-90bb3ccc4ef0
- http://finder-frontend-pr-1333.herokuapp.com/drug-device-alerts
- http://finder-frontend-pr-1333.herokuapp.com/find-eu-exit-guidance-business
- http://finder-frontend-pr-1333.herokuapp.com/find-eu-exit-guidance-business?keywords=eori&order=relevance
- http://finder-frontend-pr-1333.herokuapp.com/find-eu-exit-guidance-business?sector_business_area%5B%5D=aerospace&order=topic
- http://finder-frontend-pr-1333.herokuapp.com/uk-nationals-living-eu
- http://finder-frontend-pr-1333.herokuapp.com/prepare-business-uk-leaving-eu

[Other finders](https://live-stuff.herokuapp.com/finders)

### CODE REVIEW

File: app/presenters/entry_presenter.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-           :summary,
            :path,
            to: :entry
 
   WEBSITE_ROOT = Plek.current.website_root.gsub(/https?:\/\//, '')
 
-  def initialize(entry)
+  def initialize(entry, show_summaries)
     @entry = entry
+    @show_summaries = show_summaries
```
Comment (sihugh) [inline app/presenters/entry_presenter.rb]: Are there some tests that exercise this somewhere? Atom feeds seem like a sensitive thing

File: app/models/document.rb [hunk 2]
```diff
@@ -38,10 +37,6 @@ def truncated_description
     description.gsub(/\.\s[A-Z].*/, '.') if description.present?
   end
 
-  def summary
```
Comment (sihugh) [inline app/models/document.rb]: Was this only used by atom feeds? 😮

Comment (koetsier) [reply app/models/document.rb]: I searched the codebase for 'summary' and Document#summary seems to only be used in EntryPresenter#summary.

A very similar method lives in SearchResultPresenter#summary_text which uses Document#truncated_description directly.

It's not great that there are two methods in different places doing similar things but this is just one small step in many refactoring to hopefully make things better

File: app/presenters/entry_presenter.rb (hunk 1 above)
Comment (koetsier) [reply app/presenters/entry_presenter.rb]: The atom feeds do not have a great test suite. EntryPresenter does not have its own unit tests - it is tested a bit by the feature tests. I'll add some unit tests

File: app/models/document.rb [hunk 3]
```diff
@@ -91,20 +95,20 @@ def build_date_metadata(key)
     }
   end
 
-  def tag_metadata
-    tag_metadata_keys
-      .map(&method(:build_tag_metadata))
+  def text_metadata
+    text_metadata_keys
+      .map { |key| build_text_metadata(key) }
       .select(&method(:metadata_value_present?))
```
[the remark this answers is not in the document]
Comment (koetsier) [reply app/models/document.rb]: 👍

**Comment by koetsier** [comment]
rebased

**Pull request merged**